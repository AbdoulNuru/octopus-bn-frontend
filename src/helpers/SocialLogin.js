import React, { Component } from 'react';
import queryString from 'query-string';
import { LinearProgress } from '@material-ui/core';
import { loginUserSuccess } from '../redux/actions/loginAction';
import { connect } from 'react-redux';
class SocialLogin extends Component {
	constructor(props) {
		super(props);
	}
	componentDidMount() {
		try {
			const values = queryString.parse(this.props.location.search);
			const data = values.data;
			const user = JSON.parse(data)[0];
			const token = JSON.parse(data)[2];
			localStorage.setItem('bn-user-data', JSON.stringify({ data: user, token }));
			this.props.loginUserSuccess({ data: user, token });
			window.location.href = `/dashboard`;
		} catch (error) {
			window.location.href = `/lost`;
		}
	}
	render() {
		return (
			<div>
				<LinearProgress />
			</div>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		loginSate: state.login
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		loginUserSuccess: (user) => dispatch(loginUserSuccess(user))
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(SocialLogin);
