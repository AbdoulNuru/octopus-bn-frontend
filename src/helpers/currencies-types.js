export const currencies = [
  {
    value: "RWF",
    label: "RWF",
  },
  {
    value: "USD",
    label: "USD $",
  },
  {
    value: "EUR",
    label: "EURO â‚¬",
  },
];

export const types = [
  {
    value: "Single",
    label: "Single",
  },
  {
    value: "Double",
    label: "Double",
  },
  {
    value: "Triple",
    label: "Triple",
  },
  {
    value: "Quad",
    label: "Quad",
  },
  {
    value: "Queen",
    label: "Queen",
  },
  {
    value: "King",
    label: "King",
  },
  {
    value: "Twin",
    label: "Twin",
  },
  {
    value: "Double-double",
    label: "Double double",
  },
  {
    value: "Studio",
    label: "Studio",
  },
  {
    value: "Master-Suite",
    label: "Master suite",
  },
  {
    value: "Mini-Suite",
    label: "Mini suite",
  },
];
