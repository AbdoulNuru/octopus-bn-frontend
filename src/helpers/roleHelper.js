import axios from "axios";

export default {
  admin: {
    roleSettings: (data) =>
      axios
        .post(
          "https://octopus-bn-backend.herokuapp.com/api/v1/auth/assign-roles",
          {
            email: data.email,
            role: data.role,
          },
          {
            headers: {
              "Content-Type": "application/json",
              "x-access-token": `${localStorage.getItem("bn-token")}`,
            },
          }
        )
        .then((res) => res.data.user),
    addAccommodation: (data) =>
      axios
        .post(
          "https://octopus-bn-backend.herokuapp.com/api/v1/accommodations",
          {
            name: data.name,
            city: data.city,
            country: data.country,
            city: data.city,
            imageUrl: data.imageUrl,
            amenities: data.amenities,
            around: data.around,
          },
          {
            headers: {
              "Content-Type": "application/json",
              "x-access-token": `${localStorage.getItem("bn-token")}`,
            },
          }
        )
        .then((res) => res.data.accommodation),
    addRooms: (data) =>
      axios
        .post(
          "https://octopus-bn-backend.herokuapp.com/api/v1/accommodations/room",
          {
            accommodationsID: data.accommodationsID,
            roomNumber: data.roomNumber,
            cost: data.cost,
            currency: data.currency,
            type: data.type,
          },
          {
            headers: {
              "Content-Type": "application/json",
              "x-access-token": `${localStorage.getItem("bn-token")}`,
            },
          }
        )
        .then((res) => res.data.room),
  },
};
