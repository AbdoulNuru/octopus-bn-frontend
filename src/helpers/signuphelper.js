import axios from "axios";

export default {
  user: {
    signup: (user) =>
      axios
        .post("https://octopus-bn-backend.herokuapp.com/api/v1/auth/signup", {
          firstName: user.firstName,
          lastName: user.lastName,
          email: user.email,
          password: user.password,
        })
        .then((res) => res.data.user),
  },
};
