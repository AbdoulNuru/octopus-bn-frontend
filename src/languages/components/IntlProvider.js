import React, { Component } from 'react';
import { IntlProvider } from 'react-intl';
import { connect } from 'react-redux';

class Provider extends Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<IntlProvider locale={this.props.languageState.locale} messages={this.props.languageState.messages}>
				{this.props.children}
			</IntlProvider>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		languageState: state.language
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		setLanguage: (language) => dispatch(loginAction(language))
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Provider);
