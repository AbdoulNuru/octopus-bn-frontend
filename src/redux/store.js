import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import rootReducer from './reducers';
import logger from 'redux-logger';
const middlwares = applyMiddleware(thunk, promise, logger);
const store = createStore(rootReducer, {}, composeWithDevTools(middlwares));

export default store;
