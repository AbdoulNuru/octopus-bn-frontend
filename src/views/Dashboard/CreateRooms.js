import React, { Component } from "react";
import { connect } from "react-redux";
import entry from "../../styles/entry.module.css";
import title from "../../styles/addAccom.module.css";
import CreateRooms from "../../components/CreateRoomsForm";
import { addRoom } from "../../redux/actions/addRoomAction";

class createRoomsPage extends Component {
  submit = (data) =>
    this.props.addRoom(data).then(() => this.props.history.push("/add-rooms"));
  render() {
    return (
      <div className={entry.container}>
        <h1 className={title.title}>Add rooms</h1>
        <p className={title.sub_title}>Travel made easy - Barefoot Nomad</p>
        <CreateRooms submit={this.submit} />
      </div>
    );
  }
}

export default connect(null, { addRoom })(createRoomsPage);
