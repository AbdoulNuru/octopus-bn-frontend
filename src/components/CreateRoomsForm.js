import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import axios from "axios";
import MenuItem from "@material-ui/core/MenuItem";
import CircularProgress from "@material-ui/core/CircularProgress";
import Snackbar from "@material-ui/core/Snackbar";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";
import PropTypes from "prop-types";
import { currencies, types } from "../helpers/currencies-types";
import styles from "../styles/signup.module.css";
import validate from "../helpers/validateRoom";

class createRooms extends Component {
  state = {
    accommodationsID: "",
    roomNumber: "",
    cost: "",
    currency: "",
    type: "",
    errors: {},
    accommodations: [],
    loading: "none",
    open: false,
    message: "",
    btnMessage: "Add",
  };

  onChange = (e) =>
    this.setState({
      [e.target.name]: e.target.value,
      message: "",
    });

  onSubmit = (e) => {
    e.preventDefault();
    const errors = validate(
      this.state.accommodationsID,
      this.state.roomNumber,
      this.state.cost,
      this.state.currency,
      this.state.type
    );
    this.setState({ errors });
    if (Object.keys(errors).length === 0) {
      const data = {
        accommodationsID: this.state.accommodationsID,
        roomNumber: this.state.roomNumber,
        cost: this.state.cost,
        currency: this.state.currency,
        type: this.state.type,
      };
      this.setState({ loading: "block" });
      this.props
        .submit(data)
        .then(() => {
          this.setState({
            message: "The room was successfully registered",
            loading: "none",
          });
        })
        .catch((err) =>
          this.setState({
            errors: { globalError: err.response.data.error },
            loading: "none",
            open: true,
          })
        );
    }
  };

  componentDidMount() {
    const token = localStorage.getItem("bn-token");
    const header = {
      headers: {
        "Content-Type": "application/json",
        "x-access-token": `${token}`,
      },
    };
    axios
      .get(
        "https://octopus-bn-backend.herokuapp.com/api/v1/accommodations/all-accommodations",
        header
      )
      .then((res) => {
        this.setState({ accommodations: res.data.data.allAccommodations });
      });
  }

  render() {
    const {
      errors,
      accommodations,
      loading,
      btnMessage,
      open,
      message,
    } = this.state;
    console.log(accommodations.length);
    return (
      <div className={styles.forms_plus_second}>
        <div className={styles.forms}>
          <div className={styles.third_wrap}>
            <form
              className={styles.form}
              aria-label="form"
              onSubmit={this.onSubmit}
            >
              <TextField
                className={styles.inputs}
                label="Accommodation"
                inputProps={{
                  "aria-label": "Accommodation",
                }}
                select
                name="accommodationsID"
                variant="outlined"
                error={errors.accommodationErrorStatus}
                helperText={errors.accommodation}
                value={message === "" ? this.state.accommodationsID : ""}
                onChange={this.onChange}
              >
                {accommodations.length === 0 ? (
                  <CircularProgress style={{ position: "center" }} />
                ) : (
                  accommodations.map((info) => (
                    <MenuItem key={info.name} value={info.id}>
                      {info.name}
                    </MenuItem>
                  ))
                )}
              </TextField>
              <TextField
                className={styles.inputs}
                style={{ marginTop: "10px" }}
                label="Room number"
                inputProps={{
                  "aria-label": "room number",
                }}
                name="roomNumber"
                variant="outlined"
                error={errors.roomErrorStatus}
                helperText={errors.room}
                value={message === "" ? this.state.roomNumber : ""}
                onChange={this.onChange}
              />
              <TextField
                className={styles.inputs}
                style={{ marginTop: "10px" }}
                label="Cost"
                inputProps={{
                  "aria-label": "cost",
                }}
                name="cost"
                variant="outlined"
                error={errors.costErrorStatus}
                helperText={errors.cost}
                value={message === "" ? this.state.cost : ""}
                onChange={this.onChange}
              />
              <TextField
                className={styles.inputs}
                style={{ marginTop: "10px" }}
                label="Currency"
                inputProps={{
                  "aria-label": "currency",
                }}
                select
                name="currency"
                variant="outlined"
                error={errors.currencyErrorStatus}
                helperText={errors.currency}
                value={message === "" ? this.state.currency : ""}
                onChange={this.onChange}
              >
                {currencies.map((currency) => (
                  <MenuItem key={currency.label} value={currency.value}>
                    {currency.label}
                  </MenuItem>
                ))}
              </TextField>
              <TextField
                className={styles.inputs}
                style={{ marginTop: "10px" }}
                label="Type"
                inputProps={{
                  "aria-label": "type",
                }}
                select
                name="type"
                variant="outlined"
                error={errors.typeErrorStatus}
                helperText={errors.type}
                value={message === "" ? this.state.type : ""}
                onChange={this.onChange}
              >
                {types.map((type) => (
                  <MenuItem key={type.label} value={type.value}>
                    {type.label}
                  </MenuItem>
                ))}
              </TextField>
              <Button
                type="submit"
                aria-label="submit"
                variant="contained"
                color="primary"
                style={{
                  float: "right",
                  marginTop: "10px",
                  textTransform: "capitalize",
                  width: "136px",
                }}
                disableElevation
              >
                {loading === "block" ? (
                  <div className={styles.progress}>
                    <CircularProgress size={24} color="#4b7ba0" />
                  </div>
                ) : (
                  btnMessage
                )}
              </Button>
            </form>
            <p style={{ color: "green" }}>{message}</p>
          </div>
        </div>
        <Snackbar
          anchorOrigin={{
            vertical: "bottom",
            horizontal: "right",
          }}
          open={open}
          autoHideDuration={6000}
          onClose={this.handleClose}
          ContentProps={{
            "aria-describedby": "message-id",
          }}
          message={<span id="message-id">{errors.globalError}</span>}
          action={[
            <IconButton
              key="close"
              aria-label="Close"
              color="inherit"
              onClick={() => this.setState({ open: false })}
            >
              <CloseIcon />
            </IconButton>,
          ]}
        />
      </div>
    );
  }
}

createRooms.propTypes = {
  submit: PropTypes.func.isRequired,
};

export default withRouter(createRooms);
