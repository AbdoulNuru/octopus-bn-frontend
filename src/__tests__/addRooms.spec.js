import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Link } from "react-router-dom";
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import store from "../redux/store";
import IntlProvider from "../languages/components/IntlProvider";
import CreateRooms from "../views/Dashboard/CreateRooms";
import CreateRoomsForm from "../components/CreateRoomsForm";
import { currencies } from "../helpers/currencies-types";

const AddRooms = () => {
  return render(
    <Provider store={store}>
      <IntlProvider>
        <Router>
          <CreateRooms />
        </Router>
      </IntlProvider>
    </Provider>
  );
};

describe("Travel admin", () => {
  addEventListener;
  afterEach(cleanup);

  it("should render create rooms page", () => {
    const { asFragment } = AddRooms();
    expect(asFragment(<CreateRooms />)).toMatchSnapshot();
  });

  it("should render create rooms form", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <IntlProvider>
            <CreateRoomsForm />
          </IntlProvider>
        </Router>
      </Provider>
    );
    expect(asFragment(<CreateRoomsForm />)).toMatchSnapshot();
  });

  it("should be able to deal with data change", () => {
    const { getByLabelText } = AddRooms();
    const accommodationsID = getByLabelText("Accommodation");
    const roomNumber = getByLabelText("room number");
    const cost = getByLabelText("cost");
    const currency = getByLabelText("currency");

    userEvent.selectOptions(accommodationsID, "Qwhehdbh");
    userEvent.type(roomNumber, "12H");
    userEvent.type(cost, "12");
    userEvent.selectOptions(currency, "RWF");

    expect(accommodationsID.value).toBe(undefined);
    expect(roomNumber.value).toBe("12H");
    expect(cost.value).toBe("12");
    expect(currency.value).toBe(undefined);
  });

  it("should show an error when there is no accommodation provided", () => {
    const { getByLabelText, getByText, container } = AddRooms();
    const accommodationsID = getByLabelText("Accommodation");
    const roomNumber = getByLabelText("room number");
    const cost = getByLabelText("cost");
    const form = container.querySelector("form");

    userEvent.type(accommodationsID, "");
    userEvent.type(roomNumber, "12H");
    userEvent.type(cost, "12");
    form.dispatchEvent(new Event("submit"));
    waitFor(() =>
      expect(getByText("You must specify the accommodation")).toBeTruthy()
    );
  });

  it("should show an error when there is no room number provided", () => {
    const { getByLabelText, getByText, container } = AddRooms();
    const accommodationsID = getByLabelText("Accommodation");
    const roomNumber = getByLabelText("room number");
    const cost = getByLabelText("cost");
    const form = container.querySelector("form");

    userEvent.type(accommodationsID, "");
    userEvent.type(roomNumber, "");
    userEvent.type(cost, "12");
    form.dispatchEvent(new Event("submit"));
    waitFor(() =>
      expect(getByText("The room number is required")).toBeTruthy()
    );
  });
});
