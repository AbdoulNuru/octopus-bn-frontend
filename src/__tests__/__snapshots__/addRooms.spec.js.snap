// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Travel admin should render create rooms form 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <form
          aria-label="form"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Accommodation
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-label="Accommodation"
                aria-labelledby="mui-component-select-accommodationsID"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="mui-component-select-accommodationsID"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                name="accommodationsID"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-216 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-218"
                >
                  <span>
                    Accommodation
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="margin-top: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Room number
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                aria-label="room number"
                class="MuiInputBase-input MuiOutlinedInput-input"
                name="roomNumber"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-216 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-218"
                >
                  <span>
                    Room number
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="margin-top: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Cost
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                aria-label="cost"
                class="MuiInputBase-input MuiOutlinedInput-input"
                name="cost"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-216 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-218"
                >
                  <span>
                    Cost
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="margin-top: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Currency
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-label="currency"
                aria-labelledby="mui-component-select-currency"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="mui-component-select-currency"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                name="currency"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-216 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-218"
                >
                  <span>
                    Currency
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="margin-top: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Type
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-label="type"
                aria-labelledby="mui-component-select-type"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="mui-component-select-type"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                name="type"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-216 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-218"
                >
                  <span>
                    Type
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            aria-label="submit"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation"
            style="float: right; margin-top: 10px; text-transform: capitalize; width: 136px;"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Add
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
        <p
          style="color: green;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Travel admin should render create rooms page 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Add rooms
    </h1>
    <p>
      Travel made easy - Barefoot Nomad
    </p>
    <div>
      <div>
        <div>
          <form
            aria-label="form"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Accommodation
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-label="Accommodation"
                  aria-labelledby="mui-component-select-accommodationsID"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="mui-component-select-accommodationsID"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  name="accommodationsID"
                  type="hidden"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-82 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-84"
                  >
                    <span>
                      Accommodation
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="margin-top: 10px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Room number
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  aria-label="room number"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  name="roomNumber"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-82 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-84"
                  >
                    <span>
                      Room number
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="margin-top: 10px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Cost
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  aria-label="cost"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  name="cost"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-82 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-84"
                  >
                    <span>
                      Cost
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="margin-top: 10px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Currency
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-label="currency"
                  aria-labelledby="mui-component-select-currency"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="mui-component-select-currency"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  name="currency"
                  type="hidden"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-82 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-84"
                  >
                    <span>
                      Currency
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="margin-top: 10px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Type
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-label="type"
                  aria-labelledby="mui-component-select-type"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="mui-component-select-type"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  name="type"
                  type="hidden"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-82 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-84"
                  >
                    <span>
                      Type
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              aria-label="submit"
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation"
              style="float: right; margin-top: 10px; text-transform: capitalize; width: 136px;"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Add
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
          <p
            style="color: green;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
